compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectPaths.java

```diff
@@ -49,7 +49,7 @@ public static Path packageRoot(Path filepath) throws ProjectException {
         }
 
         // check if the file is a regular file
-        if (!Files.isRegularFile(filepath)) {
+        if (!Files.isRegularFile(filepath) && !Files.isDirectory(filepath)) {
```

Comment:
With this change, I suppose now we support directory paths as well. So should we consider changing the parameter name + java docs accordingly?